src/main/java/frc/robot/commands/GoToNodePID.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.controls.Operator;
+import frc.robot.subsystems.Drivetrain;
+
+/**
+ * Uses PID to make the robot go to the selected pose
+ */
+public class GoToNodePID extends CommandBase {
+
+  private final Drivetrain m_drive;
+  private final Operator m_operator;
+  public GoToNodePID(Operator operator, Drivetrain drive) {
+    m_drive = drive;    
+    m_operator=operator;
+    addRequirements(drive);
+  }
+
+  @Override
+  public void initialize(){
+    m_drive.setAllOptimize(true);
+    m_drive.getXController().reset();
+    m_drive.getYController().reset();
+    m_drive.getRotationController().reset();
+  }
+  
+  @Override
+  public void execute() {
+    double x = m_operator.getSelectedNode().scorePose.getX();
+    double y = m_operator.getSelectedNode().scorePose.getY();
+    double rot = m_operator.getSelectedNode().scorePose.getRotation().getRadians();
```

Comment:
for safety, and making it more readible, you can make a m_scorePose set to to the operator's control in init().

Reply:
If it was going in the wrong direction, I wanted the operator to be able to change the selected node and make the robot go there instead. I didn't add it to the other GoToNode because the PathPlanner command can't be recreated every frame.